src/components/DepositWidget/useRowActions.tsx

```diff
@@ -47,11 +47,15 @@ export const useRowActions = (params: Params): Result => {
   }
 
   async function enableToken(tokenAddress: string): Promise<void> {
-    const { symbol } = getToken('address', tokenAddress, balances) as Required<TokenBalanceDetails>
-
     try {
-      assert(contractAddress, 'Contract address missing. Aborting.')
       assert(userAddress, 'User address missing. Aborting.')
+      assert(contractAddress, 'Contract address missing. Aborting.')
+
+      const token = getToken('address', tokenAddress, balances)
+      assert(token, 'No token, aborting.')
+
+      const { symbol: tokenDisplayName } = safeFilledToken(token)
```

Comment:
Btw, completely unnecessary here as you employ `safeFilledToken` for that, but
a nifty trick to get `tokenDisplayName = name || symbol || address` out of a token is
```ts
const {
  name: tokenDisplayName,
  symbol: tokenDisplayName = tokenDisplayName,
  address: tokenDisplayName = tokenDisplayName
} = token
```

Cool?

Reply:
hmmm super cool, what the heck

Reply:
```
let token = {
	name: undefined,
	symbol: undefined,
	address: '0x12312312312',
}

const {
  name: tokenDisplayName,
  symbol: tokenDisplayName = tokenDisplayName,
  address: tokenDisplayName = tokenDisplayName
} = token
VM313:9 Uncaught SyntaxError: Identifier 'tokenDisplayName' has already been declared
```

didnt work para mi :(

Reply:
May I suggest safeTokenName from https://github.com/gnosis/dex-react/blob/develop/src/utils/format.ts#L133 ?

Reply:
Agree with Leandro. Looks to me better handled, easier to read, and we don't copy paste this logic around.

Reply:
> didnt work para mi :(

Reply:
I prefer my implementation as I get both the name and symbol back with the token object and I don't need to do an extra param and making it one bigger complicated function.

Reply:
```js
let tokenDisplayName;
{
  name: tokenDisplayName,
  symbol: tokenDisplayName = tokenDisplayName,
  address: tokenDisplayName = tokenDisplayName
} = token
```